### PR TITLE
Reduce amout of dependencies, support python 3.10

### PR DESCRIPTION
**Pull request by eicca**
- remove unused dependencies
- create new "notebooks" extras
- move plotting and jupyter dependencies into "notebooks" extras
- support python 3.10
- remove unused requirements.txt
- relax versions of dependencies
- remove conda usage from the README

### CODE REVIEW

**Review by oalifdal — APPROVED** [review]

**Comment by oalifdal** [comment]
Nice, thanks @eicca !

File: README.md [hunk 1]
```diff
@@ -27,27 +27,21 @@ Checking how many of these guesses are correct, the success rates of the differe
 
 ## Setup and installation
 
+Anonymeter requires Python 3.8.x, 3.9.x or 3.10.x (this limitation is due to Numba usage).
```
Comment (nicola-v) [inline README.md:30]: Is the limitation about not being able to run with Python 3.11? I'd be more explicit about that (e.g., “Python 3.11 is not supported (yet) due to Numba usage”).

File: README.md [hunk 2]
```diff
@@ -75,19 +69,14 @@ risk = evaluator.risk()
 ## Getting started
```
Comment (nicola-v) [inline README.md:69]: I'd put “Getting started” just after “Setup and installation”. In fact, the “Basic usage pattern” is in the notebook, too.

**Review by nicola-v — APPROVED** [review]
Looks good to me! I have left just a couple of suggestions.

File: README.md (hunk 1 above)
Comment (eicca) [reply README.md:30]: It's also the limitation of not being able to run with Python 3.7 :) Numba supports only 3.8, 3.9 and 3.10 in the current release

Comment (eicca) [reply README.md:30]: I can just remove the numba mention to avoid confusion

**Comment by eicca** [comment]
Thanks @oalifdal and @nicola-v !

**Pull request merged**